src/data/getGenres.js

```diff
@@ -0,0 +1,11 @@
+export const getGenres = (movies) => {
+  let genresArray = [];
+  movies?.map((movie) => {
+    return movie?.genre_ids.map((genre) => {
+      if (!genresArray.includes(genre)) {
+        genresArray.push(genre);
+      }
+    });
+  });
```

Comment:
`.map()` and `.filter()` return arrays. Here you are using `.map()` as if it were a `.forEach()` method.

Your code tries to find a unique set of genre ids. For this you can use a JavaScript `Set` object which guarantees that its members are unique. Since you want to return an array instead of a Set you can spread out the Set into an array.

 

```suggestion
export const getGenres = (movies) => {
  const genreSet = new Set();
  movies?.forEach((movie) => {
    movie.genre_ids.forEach((genre_id) => genreSet.add(genre_id));
  });
  return [...genreSet];
};

```